### PR TITLE
No Issue: Fix Lint Issues Downloads Manager

### PR DESCRIPTION
**Pull request by kglazko**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by kglazko** [comment]
@liuche @csadilek Created this issue to address the lint failures. They weren't failing on the original PR and weren't failing locally, so maybe something is wrong with detekt? This passes on ktlint and detekt locally for me, so please verify that this passes on whatever CI is run when merged.

**Comment by jonalmeida** [comment]
I came here to fix an issue about this. 😄  This is full report I'm seeing from `detekt` if that's helpful.

```
536 kotlin files were analyzed.
/Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadAdapter.kt - 5min debt
        AbsentOrWrongFileLicense - [DownloadAdapter.kt] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadAdapter.kt:1:1
/Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadController.kt - 5min debt
        AbsentOrWrongFileLicense - [DownloadController.kt] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadController.kt:1:1
/Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadView.kt - 5min debt
        AbsentOrWrongFileLicense - [DownloadView.kt] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadView.kt:1:1
/Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt - 5min debt
        MaxLineLength - [DownloadItem] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:19:1

Overall debt: 20min

comments - 15min debt
        AbsentOrWrongFileLicense - [DownloadAdapter.kt] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadAdapter.kt:1:1
        AbsentOrWrongFileLicense - [DownloadController.kt] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadController.kt:1:1
        AbsentOrWrongFileLicense - [DownloadView.kt] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadView.kt:1:1
style - 5min debt
        MaxLineLength - [DownloadItem] at /Users/jalmeida/src/fenix-alt/app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:19:1

Overall debt: 20min

Complexity Report:
        - 61,651 lines of code (loc)
        - 45,903 source lines of code (sloc)
        - 30,854 logical lines of code (lloc)
        - 6,834 comment lines of code (cloc)
        - 6,772 McCabe complexity (mcc)
        - 4 number of total code smells
        - 14% comment source ratio
        - 219 mcc per 1,000 lloc
        - 0 code smells per 1,000 lloc

Project Statistics:
        - number of kt files: 536
        - number of packages: 95
        - number of classes: 1077
        - number of functions: 3926
        - number of properties: 3193
```

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt [hunk 1]
```diff
@@ -16,7 +16,8 @@ import mozilla.components.lib.state.Store
  * @property size The size in bytes of the download item
  * @property contentType The type of file the download is
  */
-data class DownloadItem(val id: Long, val fileName: String?, val filePath: String, val size: String, val contentType: String?)
+data class DownloadItem
+    (val id: Long, val fileName: String?, val filePath: String, val size: String, val contentType: String?)
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:20]: nit: we usually format these differently: https://github.com/mozilla-mobile/fenix/blob/52c2fdb310f5b86ce6484821b4901f85f15ee858/app/src/main/java/org/mozilla/fenix/home/HomeFragmentStore.kt#L25

**Comment by kglazko** [comment]
I addressed the nit and am waiting for CI. @jonalmeida Any idea why CI would have been passing locally and in the PR and then failing on Master?

**Comment by jonalmeida** [comment]
@kglazko the Synced Tabs change that I landed seems to have caused a conflict with the `FeatureFlags` change you made. 

Since that change is merged without CI failures (❗),  you may see a failure that this PR fixes: https://github.com/mozilla-mobile/fenix/pull/13947

Could you cherry-pick or fix them in this PR since you're addressing them in this PR? Or vice-versa, I can take these changes.

**Comment by jonalmeida** [comment]
> @kglazko the Synced Tabs change that I landed seems to have caused a conflict with the `FeatureFlags` change you made.
> 
> Since that change is merged without CI failures (❗), you may see a failure that this PR fixes: #13947
> 
> Could you cherry-pick or fix them in this PR since you're addressing them in this PR? Or vice-versa, I can take these changes.

For visibility, we're going to merge this PR since it doesn't contain the changes that cause master to break. We can then merge my PR (linked above) to fix master.

**Pull request merged**